src/Datadog.Trace.ClrProfiler.Managed/Integrations/WebRequestIntegration.cs

```diff
@@ -0,0 +1,117 @@
+        [InterceptMethod(
+            TargetAssembly = "System.Net.Requests", // .NET Core
+            TargetType = "System.Net.WebRequest")]
+        public static object GetResponse(object request)
+        {
+            var webRequest = (WebRequest)request;
+
+            if (!IsTracingEnabled(webRequest))
+            {
+                return webRequest.GetResponse();
+            }
+
+            using (var scope = CreateScope(webRequest, nameof(GetResponse)))
+            {
+                try
+                {
+                    // add distributed tracing headers
+                    webRequest.Headers.Inject(scope.Span.Context);
+
+                    return webRequest.GetResponse();
+                }
+                catch (Exception ex)
+                {
+                    scope.Span.SetException(ex);
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Instrumentation wrapper for <see cref="WebRequest.GetResponseAsync"/>.
+        /// </summary>
+        /// <param name="request">The <see cref="WebRequest"/> instance to instrument.</param>
+        /// <returns>Returns the value returned by the inner method call.</returns>
+        [InterceptMethod(
+            TargetAssembly = "System.Net",
+            TargetType = "System.Net.WebRequest")]
+        public static object GetResponseAsync(object request)
+        {
+            return GetResponseAsyncInternal((WebRequest)request);
+        }
+
+        private static async Task<WebResponse> GetResponseAsyncInternal(WebRequest request)
+        {
+            if (!IsTracingEnabled(request))
+            {
+                return await request.GetResponseAsync().ConfigureAwait(false);
+            }
+
+            using (var scope = CreateScope(request, nameof(GetResponseAsync)))
+            {
+                try
+                {
+                    // add distributed tracing headers
+                    request.Headers.Inject(scope.Span.Context);
+
+                    return await request.GetResponseAsync()
+                                        .ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    scope.Span.SetException(ex);
+                    throw;
+                }
+            }
+        }
+
+        private static bool IsTracingEnabled(WebRequest request)
+        {
+            // check if tracing is disabled for this request via http header
+            string value = request.Headers[HttpHeaderNames.TracingEnabled];
+            return !string.Equals(value, "false", StringComparison.InvariantCultureIgnoreCase);
+        }
+
+        private static Scope CreateScope(WebRequest request, string methodName)
+        {
+            string httpMethod = request.Method.ToUpperInvariant();
+            string url = request.RequestUri.OriginalString;
+            string resourceName = $"{httpMethod} {url}";
```

Comment:
Isn't this going to create very high cardinality resources?

Reply:
I'm not sure what you're suggesting. Should we use only specific parts of the URL?

Reply:
I'd say it is something that should be uniform accross tracers, but as a rule we should not put things that we know will have unbounded cardinality in the resource field (since we compute stats on them).
I've looked quickly and it seems that the nodejs tracer defaults to the [http method](https://github.com/DataDog/dd-trace-js/blob/master/src/plugins/http/client.js#L43)
while the python requests integration [doesn't seem to set a resource??](https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/contrib/requests/connection.py#L71)
Probably worth checking with others what the good practice is here.

Reply:
I thought the resource was usually the URL for http requests or the SQL query for data sources.

Reply:
For sql queries the agent removes the variable parts and standardizes them to have meaningful resources. For urls however there is unfortunately no generic way to standardize them, this is why for incoming http requests the tracer sets as resource the route or something else that makes sense for the given framework but never the raw url.

In any case setting the raw url would not make sense for most cases, for instance if you were to call a user service with a route like "/user/{id}", you would get one different resource page for each user id which would have little value.

Reply:
I see your point. I removed the raw URL. Looks like the other tracers haven't come to an agreement on whether we should use the request's domain as the resource. nodejs has a "split by domain" settings that is disabled by default. We can revisit that later.

Reply:
The commit removes it only in one place, the WebRequest integration still uses the url. I'm wondering if we should extract a function to create the scope for both integrations from all the infomation we put in here. That would ensure both integrations stay consistent.

Reply:
🤦🏽‍♂️ Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/267/commits/3f3fadbf5dabc0c762d817f7b76e29c3f97e6c34

> extract a function to create the scope for both integrations
Agreed. Abstracting some of these functionality is something that @boydc7 is working on as part of the Web Forms integration.

Reply:
Right, although a simple function like CreateHttpScope(url, method) would do the trick here.